### ISSUE TITLE
[BOM] User cannot use both `first-` AND `lastname` to filter orders

### ISSUE THREAD

**Issue by filipefurtad0** [openfoodfoundation/openfoodnetwork#10570]
## Description

After #10460 the user cannot use both `first-` AND `lastname` together to search for orders; it can only search for `firstname` OR `lastname`.

(in a way, this is a regression, because before #10460 the user could search before for both names - however, only on the results shown per page)

## Expected Behavior

User should be able to search for both `first-` AND `lastname`

## Actual Behaviour

User cannot use both `first-` AND `lastname` together to search for orders

## Steps to Reproduce

1. Log in as admin of a hub with some complete orders and visit `/admin/orders/bulk_management`
2. On the search field, type the first and last name
3. Click "Filter Results"
4. Notice no results are found.

## Animated Gif/Screenshot

Using first name only - OK

![image](https://user-images.githubusercontent.com/49817236/225380484-b7a313d9-e603-43fd-9be2-41a5c41793c6.png)

Using first and lastname - No results:

![image](https://user-images.githubusercontent.com/49817236/225380608-00c9db9c-08ee-49ef-99eb-7937b8b0e3e0.png)

## Workaround

Type in only first OR lastname.

## Severity

bug-s4: it's annoying, but you can use it

## Your Environment

* Version used: Should be in v4.2.39
* Browser name and version: Firefox 110.0.1
* Operating System and version (desktop or mobile): Ubuntu 22.04

**Comment by saunmanoj888** [comment]
Hey!!
If this issue is not picked. Can I work on this?

**Comment by sigmundpetersen** [comment]
Sure, thanks @saunmanoj888 👍 
I've assigned you

**Comment by saunmanoj888** [comment]
hi @drummer83 
Just to confirm "Azra" is the last name and "Angel" is the first name. In PR https://github.com/openfoodfoundation/openfoodnetwork/pull/10813 the fix was made only to search when a string is entered in following order `Firstname Lastname`.  Do we also want to search with `Lastname Firstname` order? Also do we want to include comma while searching with full name? as mentioned in the third case?. 

I can work on this issue again and fix unresolved test cases. Just want to confirm that we want to allow customer to search will all possible scenarios mentioned here https://github.com/openfoodfoundation/openfoodnetwork/issues/10570#issuecomment-1551264769.

**Comment by drummer83** [comment]
Thanks @saunmanoj888!

> Just to confirm "Azra" is the last name and "Angel" is the first name.

Yes, that's right.

My opinion:
- we should ignore the order of first and last name
- and we should also ignore if there is a comma or not.

But maybe we should hear some other opinions here from @openfoodfoundation/train-drivers-product-owners or @openfoodfoundation/testers.

**Comment by saunmanoj888** [comment]
> Thanks @saunmanoj888!
> 
> > Just to confirm "Azra" is the last name and "Angel" is the first name.
> 
> Yes, that's right.
> 
> My opinion:
> 
> * we should ignore the order of first and last name
> * and we should also ignore if there is a comma or not.
> 
> But maybe we should hear some other opinions here from @openfoodfoundation/train-drivers-product-owners or @openfoodfoundation/testers.

thanks @drummer83. 
Sure, I'll wait for the product owners & testers opinions as well, once the behaviour is finalized, I'll start working on this issue.

**Comment by sigmundpetersen** [comment]
@RachL any thoughts on Konrad's suggestion?

**Comment by RachL** [comment]
I think we are looking at edge cases here. The rule you give @drummer83 sounds good but lets open a new issue for this and keep this one closed? @saunmanoj888 if the implementation is too complex don't spend too much time on it. We are already solving most cases I think :-)

**Comment by drummer83** [comment]
> We are already solving most cases I think

I don't agree to this. When I tested the first PR I was surprised that my first attempts still returned no results.

Could you please explain the benefit of creating a new issue?

**Comment by RachL** [comment]
@drummer83 there is a risk this will not be finished by @saunmanoj888 as they are voluntarily contributing (and that's ok). It's better for future contributors to have a clean issue with clean acceptance criteria without having to dig into comments. Use cases you've tested are not listed in the original issue.

**Comment by RachL** [comment]
Follow-up issue created here: https://github.com/openfoodfoundation/openfoodnetwork/issues/11585
